src/framework/preferences/js/PrefsEditor.js

```diff
@@ -454,7 +458,22 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
             that.locate("cancel").click(that.cancel);
         };
 
-        that.container.append(that.resources.template.resourceText);
+        var template = that.resources.template.resourceText;
+        if (that.options.generatePanelContainers) {
+            var sorted = fluid.parsePriorityRecords(that.options.preferences, "preferences"); // sort the preferences
+            var sortedPrefs = fluid.getMembers(sorted, "namespace"); // retrieve just the preferences names
+
+            // generate panel container markup
+            var panels = fluid.accumulate(sortedPrefs, function (pref, containers) {
```

Comment:
Just a tiny bit more idiomatic is var panels = sortedPrefs.map(function(pref,containers) { ....
    return fluid.stringTemplate( ....)
}).join("");